### ISSUE TITLE
AttributeError: 'MenuItem' object has no attribute 'SetCheckable'

### ISSUE THREAD

**Issue by msuzu15** [NVDARemote/NVDARemote#176]
Hello, 

I'm using NVDA Remote 2.1 with NVDA 2018.3jp-beta-180806n (Japan localized version of NVDA) and getting following issue.  My colleague said it is due to change of version for GUI module WXPython (version 3 -> version 4) in NVDA 2018.3.  Can you please help to check if it also happen with global version of NVDA, and make change to adopt to WXPython version 4?

1. Open NVDA menu.  If NVDA Remote add-on is installed, sub menu "Remote" should be under Tools menu but it is missing.
2. Go to Tools -> Log Viewer, then I can see following error.

ERROR - globalPluginHandler.initialize (14:45:44.137):
Error initializing global plugin <class 'globalPlugins.remoteClient.GlobalPlugin'>
Traceback (most recent call last):
File "globalPluginHandler.pyo", line 32, in initialize
File "C:\Users\msuzuki\AppData\Roaming\nvda\addons\remote\globalPlugins\remoteClient_init_.py", line 56, in init
File "C:\Users\msuzuki\AppData\Roaming\nvda\addons\remote\globalPlugins\remoteClient_init_.py", line 104, in create_menu
AttributeError: 'MenuItem' object has no attribute 'SetCheckable'

3. I checked line 104 of C:\Users\msuzuki\AppData\Roaming\nvda\addons\remote\globalPlugins\remoteClient_init_.py, it is as following.

 self.mute_item.SetCheckable(True)

My colleague said the attribute "SetCheckable" is not supported in WXPython version 4 so teh code should be modified to adopt to version 4.

**Comment by msuzu15** [comment]
My colleague suggested following modification to __init__.py, I confirmed that it works with both NVDA2018.2 and NVDA2018.3.

Line 102 - 106 (original):
		# Translators: Menu item in NvDA Remote submenu to mute speech and sounds from the remote computer.
		self.mute_item = self.menu.Append(wx.ID_ANY, _("Mute remote"), _("Mute speech and sounds from the remote computer"))
		self.mute_item.SetCheckable(True)
		self.mute_item.Enable(False)
		gui.mainFrame.sysTrayIcon.Bind(wx.EVT_MENU, self.on_mute_item, self.mute_item)

Modification:

		# Translators: Menu item in NvDA Remote submenu to mute speech and sounds from the remote computer.
		# --- modification to adopt to WXPython ver.4 ---
		# original code:
		# self.mute_item = self.menu.Append(wx.ID_ANY, _("Mute remote"), _("Mute speech and sounds from the remote computer"))
		# self.mute_item.SetCheckable(True)
		# modification:
		if int(wx.version()[0]) <= 3:
		# WXPython ver. 3
			self.mute_item = self.menu.Append(wx.ID_ANY, _("Mute remote"), _("Mute speech and sounds from the remote computer"))
			self.mute_item.SetCheckable(True)
		else:
		# WXPython ver. 4
			self.mute_item = self.menu.Append(wx.ID_ANY, _("Mute remote"), _("Mute speech and sounds from the remote computer"), kind=wx.ITEM_CHECK)
		# --- end of modification to adopt to WXPython ver.4 ---
		self.mute_item.Enable(False)
		gui.mainFrame.sysTrayIcon.Bind(wx.EVT_MENU, self.on_mute_item, self.mute_item)

**Comment by leonardder** [comment]
Note that this has already been fixed in the master branch. This is a friendly but urgent request to @ctoth or @tspivey to update the build on nvdaremote.com.

**Comment by leonardder** [comment]
see also nvaccess/nvda#7104

**Comment by msuzu15** [comment]
OK, thanks.

**Comment by msuzu15** [comment]
Since the issue could be critical as it disables NVDA Remote entirely with NVDA2018.3, I hope the fix will be released soon.

**Comment by ctoth** [comment]
Fixed in #170
